packages/ui-components/src/components/Chat.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { useState } from 'react';
+import { Typography, Input } from '@material-ui/core';
+import styled, { keyframes } from 'styled-components';
+import ArrowUpwardRoundedIcon from '@mui/icons-material/ArrowUpwardRounded';
+import UndoRoundedIcon from '@mui/icons-material/UndoRounded';
+
+import { IconButton } from './IconButton';
+import { TypingIndicator } from './TypingIndicator';
+
+// Types
+interface Message {
+  id: number | string;
+  text: string;
+  isOwn: boolean;
+}
+
+interface ChatProps {
+  startingMessageText?: string;
+  messages?: Message[];
+  onSendMessage?: (message: Message) => void;
+  height?: number;
+  width?: number;
+  isProcessingMessage?: boolean;
+}
+
+interface StyledProps {
+  width: number;
+  height: number;
+}
+
+interface MessageWrapperProps {
+  isOwn: boolean;
+}
+
+interface MessageBubbleProps {
+  isOwn: boolean;
+}
+
+interface MessageContentProps {
+  isOwn: boolean;
+  bold: boolean;
+}
+
+interface ThemeProps {
+  theme: {
+    palette: {
+      background: {
+        grey: string;
+      };
+      primary: {
+        white: string;
+        blue: string;
+      };
+      primaryDark: string;
+    };
+  };
+}
+
+// Keyframe animations
+const fadeIn = keyframes`
+  from { 
+    opacity: 0; 
+    transform: translateY(10px); 
+  }
+  to { 
+    opacity: 1; 
+    transform: translateY(0); 
+  }
+`;
+
+// Styled Components
+const ChatContainer = styled.div<StyledProps>`
+  width: ${props => props.width}px;
+  height: ${props => props.height}px;
+  display: flex;
+  flex-direction: column;
+  overflow: hidden;
+  font-family: 'Roboto', 'Arial', sans-serif;
+`;
+
+const MessagesArea = styled.div<ThemeProps>`
+  flex: 1;
+  overflow-y: auto;
+  background: ${props => props.theme.palette.background.grey};
+  display: flex;
+  flex-direction: column;
+  gap: 16px;
+`;
+
+const MessageWrapper = styled.div<MessageWrapperProps>`
+  display: flex;
+  align-items: flex-end;
+  gap: 8px;
+  animation: ${fadeIn} 0.3s ease-in;
```

Comment:
Nice touch. For animating in, an ease out curve will feel more natural. (Rule of thumb is to ease in when animating out, and ease out when animating in.)

```suggestion
  animation: ${fadeIn} 0.3s --ease-out-quad;
```

In `packages/datatrak-web/src/theme/theme.ts`, you’ll find a set of timing functions that move a bit more gracefully. (The built-ins are a bit clunky.) Might be time to copy these over to **admin-panel**’s `theme.js`:

```js
MuiCssBaseline: {
  ':root': {
        '--ease-in-quad': 'cubic-bezier(0.11, 0, 0.5, 0)',
        '--ease-out-quad': 'cubic-bezier(0.5, 1, 0.89, 1)',
        // ...et al
  }
}
```